xla/stream_executor/device_description.h

```diff
@@ -54,13 +57,45 @@ class RocmComputeCapability {
 
   std::string gcn_arch_name() const { return gcn_arch_name_; }
 
+  std::string ToString() const { return gcn_arch_name(); }
+
+  RocmComputeCapabilityProto ToProto() const {
+    RocmComputeCapabilityProto proto;
+    proto.set_gcn_arch_name(gcn_arch_name_);
+    return proto;
+  }
+
+  bool operator==(const RocmComputeCapability &other) const {
+    return gcn_arch_name_ == other.gcn_arch_name_;
+  }
+
   std::string gfx_version() const {
-    std::vector<std::string> tokens = absl::StrSplit(gcn_arch_name_, ':');
-    return tokens[0];
+    //  std::strchr() is faster for the case than std::string::find()
+    const char *const p_colon = std::strchr(gcn_arch_name_.c_str(), ':');
+    if (nullptr == p_colon) {
+      return gcn_arch_name_;  // likely it's the default invalid value
+    }
```

Comment:
How is it relevant that this utility method is fast?

If this needs to be fast then we should do the split in the constructor and keep both parts around separately.

If you just want to make a bit faster we can avoid some temporary allocations:

```
std::pair<absl::string_view, absl::string_view> tokens = absl::StrSplit(gcn_arch_name_, ':');
return std::string(tokens.first);
```

But I'm a bit opposed to using raw C pointers and C library functions which makes the code harder to read.